### PR TITLE
Break up channels material with more exercises.

### PR DESCRIPTION
**Pull request by jcbwlkr**
TLDR: Channel section is too long. Would like feedback on 2 new exercises. Slightly reordering the order I discuss channel mechanics.

I want to redo the flow of the Channels section. This is based on my own opinion as well as feedback from multiple students.

So in general I try to teach for at most 20 minutes then do an exercise. This works for most content of most days until I get to Channels. That one has always just been *long*.

Here is how I would like to teach channels moving forward.

# Channels Part 1
Learn how to make channels, send on a channel, receive from a channel, make a buffered channel, and receiving in a loop. We will cover:

- `waitForTask()`
- `waitForResult()`
- `fanOut()`

Then we will do Exercise 2 which launches 100 goroutines and collects their results.

# Channels Part 2

Learn how to close a channel, receive with 2 variables, use range to iterate over a channel, and the first mention of goroutine leaks. We will cover:

- `waitForFinished()`
- `poolingUnlimited()`

We will then do the new Exercise 3. This is the same as Exercise 2 but it introduces some more complexity because not every goroutine ends up sending on the channel. This requires them to use `range` on the receiving end and have an extra goroutine whose only job is closing the channel.

At this point when we review Exercise 3 we will also discuss the 4th trap of concurrency: Unnecessary Complexity. I live-code the synchronous version which is about 5 lines of code and discuss if the potential performance gain is really worth all of this extra code.

# Channels Part 3

This covers using `select` cases both with and without a default case. Runs functions:

- `cancellation()`
- `drop()`

I like to do `cancellation` before `drop` because it makes more sense to me to explain who `select` works without a default case before adding it in.

To practice using select I created Exercise 4. This is a program that has the main goroutine and another goroutine communicate with each other using `select`. The main goroutine waits to receive from either the data channel or a timeout channel. If the timeout occurs then main will close a "shutdown" channel. The 2nd goroutine runs an infinite loop and tries to either send on the data channel or receive from the shutdown channel. A third "complete" channel is included to to signal completion when the goroutine shuts down cleanly.

This Exercise 4 is the part I am most keen to get feedback on. It's basically a simpler version of `example5/example5.go` that students are more likely to be able to complete in the time I alot for exercises.

### CODE REVIEW

File: topics/go/concurrency/channels/example1/example1.go [hunk 1]
```diff
@@ -118,12 +131,12 @@ func poolingUnlimited() {
 	fmt.Println("-------------------------------------------------------------")
 }
 
-// poolingLimited: Think about being a manager and hiring a team of employees.
+// fanOutLimited: Think about being a manager and hiring a team of employees.
```
Comment (ardan-bkennedy) [inline topics/go/concurrency/channels/example1/example1.go]: Description is wrong and this is not in the right order as the ordering in main.
